src/lib/util.js

```diff
@@ -111,6 +116,63 @@ function createDownloadIframe() {
     return iframe;
 }
 
+/**
+ * Checks if the url is a download host, but not the default download host.
+ *
+ * @public
+ * @param {string} url - The URL to check
+ * @return {boolean} - HTTP response
+ */
+export function isNonDefaultDownloadHost(url) {
+    return !url.startsWith(DEFAULT_DOWNLOAD_HOST_PREFIX) && !!url.match(/^https:\/\/dl\d+\./);
+}
+
+/**
+ * Replaces the hostname of a download URL with the default hostname, https://dl.
+ *
+ * @public
+ * @param {string} url - The URL to modify
+ * @return {string} - The updated download URL
+ */
+export function replaceDownloadHostWithDefault(url) {
+    return url.replace(/^https:\/\/dl\d+\./, DEFAULT_DOWNLOAD_HOST_PREFIX);
+}
+
+/**
+ * Sets session storage to use the default download host
+ *
+ * @public
+ * @return {void}
+ */
+export function setDownloadHostFallback() {
+    sessionStorage.setItem(DOWNLOAD_HOST_FALLBACK_KEY, 'true');
+}
+
+/**
+ * Stores the host in an array via localstorage so that we don't show a notification for it again
+ *
+ * @public
+ * @param {string} downloadHost - Download URL host
+ * @return {void}
+ */
+export function setDownloadHostNotificationShown(downloadHost) {
```

Comment:
`cacheDownloadHostNotificationShown`?

Reply:
It's a little more extreme than a cache since we are using localStorage, but I'll think some more about the name.